frontend/src/app/modules/availability/new-slot/schedule-week/schedule-week.component.ts

```diff
@@ -35,15 +35,14 @@ export class ScheduleWeekComponent extends BaseComponent implements OnInit {
     }
 
     eventTimesChanged(changedEvent: CalendarEventTimesChangedEvent): void {
-        changedEvent.event.start = changedEvent.newStart;
-        changedEvent.event.end = changedEvent.newEnd;
+        changedEvent.event.start = this.dateTimeFloor(changedEvent.newStart);
+        changedEvent.event.end = this.dateTimeFloor(changedEvent.newEnd!);
+
         const item = this.items.find(i => i.weekDay === changedEvent.event.id);
 
         if (item) {
-            item.start = `${this.reformat(changedEvent.newStart.getHours())}:${this.reformat(changedEvent.newStart.getMinutes())}:00`;
-            if (changedEvent.newEnd) {
-                item.end = `${this.reformat(changedEvent.newEnd.getHours())}:${this.reformat(changedEvent.newEnd.getMinutes())}:00`;
-            }
+            item.start = `${this.reformat(changedEvent.event.start.getHours())}:${this.reformat(changedEvent.event.start.getMinutes())}:00`;
```

Comment:
Lines 44 and 45 duplicates. I think you can create separate method/helper and use it